### ISSUE TITLE
Crash with React Native

### ISSUE THREAD

**Issue by henrikra** [jaredpalmer/formik#916]
Reopened #894 because it was closed by mistake

@jaredpalmer Please try out the branch and look the picture of the error I took. As you can see the crash wasn't because of wrong documentation. So please don't close this before validating :)

## Current Behavior
When using Formik with React Native the app will crash. Here is minimal code for reproduction https://github.com/henrikra/formikTest/blob/master/App.js
![image](https://user-images.githubusercontent.com/12229968/45352664-acf73d80-b5c1-11e8-9309-19f0503e0b96.png)

## Steps to Reproduce
Make fresh React Native project and add the code from example: https://jaredpalmer.com/formik/docs/guides/react-native

## Expected behavior
I expect the app not to crash when using Formik

## Suggested solution(s)
I have no idea how to solve this at the moment. Looks like React Native doesn't have `context` object?

## Additional context
Here is minimal reposity to reproduce this crash: https://github.com/henrikra/formikTest/blob/master/App.js

---

- Formik Version: 1.2.0
- React Native Version: 0.54.4
- TypeScript Version: 3.0.1
- OS: OSX
- Node Version: 8.11.3
- Package Manager and Version: Npm version 5.6.0

**Comment by jaredpalmer** [comment]
Can you make an expo snack to reproduce this?

**Comment by jaredpalmer** [comment]
Cc @browniefed . Any thoughts?

**Comment by henrikra** [comment]
@jaredpalmer Expo is not needed because here is the small repo that reproduces this: https://github.com/henrikra/formikTest/blob/master/App.js Link is already in the issue

**Comment by browniefed** [comment]
That particular version of React Native is running on an older version of React. Guessing some weird context resolution stuff is causing the issue then.

**Comment by henrikra** [comment]
What React version is preferred with Formik?

**Comment by browniefed** [comment]
Expo is usually running on an older React, in this case it's still running on `"react": "16.3.1"`.

Based on release notes (https://github.com/jaredpalmer/formik/releases/tag/v1.2.0) looks like things were fixed/reconfigured to work with upgraded v16.4 of React.

**Comment by browniefed** [comment]
Not 100% certain that's what we're seeing here, but that seems like a likely cause.

**Comment by jaredpalmer** [comment]
@browniefed it _should_ be backwards compat because we use `create-react-context` but idk.

**Comment by simonso-nzme** [comment]
Getting similar issue, I put the sample code in my project and it crashes right away with similar error message.

Code be related to react-native version? or react version? 

"formik": "^1.3.1",
"react": "^16.4.0",
"react-dom": "^16.4.0",
"react-native": "0.54.4",

![simulator screen shot - iphone 6 - 2018-10-05 at 16 26 47](https://user-images.githubusercontent.com/22926735/46514643-9f309180-c8bb-11e8-8f0e-baa48a8f070a.png)

**Comment by jaredpalmer** [comment]
Can you share your code?

**Comment by henrikra** [comment]
@simonso-nzme Make a small repo to github with crashing code?

**Comment by henrikra** [comment]
@jaredpalmer Did you try my repro repo https://github.com/henrikra/formikTest ?

**Comment by simonso-nzme** [comment]
@jaredpalmer @henrikra 
I checked out @henrikra's test repo and I encountered the exact same issue in my project.
My initial thought was probably something to do with the react native version v0.54.4 as both of us are using v0.54.4 and the error was throwing at ReactNativeRenderer.

So I've done a few tests, I upgraded the version to v0.56.1 on Henrik's repo and in order to make it compile I also have to upgrade "babel-preset-react-native" to "5.0.2".

so the following works:

...
  "dependencies": {
    "formik": "^1.2.0",
    "react": "16.5.0",
    "react-native": "0.56.1"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "babel-preset-react-native": "5.0.2",
    "jest": "23.6.0",
    "react-test-renderer": "16.5.0"
  ...

Now v0.54.4 to v0.56.1 is quite a big jump in terms of upgrading my project (a few breaking changes here and there; dropping iOS 8 support etc), I was hoping to see if Formik can support 'slightly' older version of react-native

Thanks!
